modules/programs/vscode.nix

```diff
@@ -47,12 +57,16 @@ in
   };
 
   config = mkIf cfg.enable {
-    home.packages = [
-      (pkgs.vscode-with-extensions.override {
-        vscodeExtensions = cfg.extensions;
-      })
-    ];
+    home.packages = [ cfg.package ];
 
     home.file."${configFilePath}".text = builtins.toJSON cfg.userSettings;
+    #adapted from https://discourse.nixos.org/t/vscode-extensions-setup/1801/2
+    home.activation.vscExtensions = dag.entryAfter ["installPackages"] ''
```

Comment:
I don't see how this implementation removes an extension from `~/.vscode/extensions` when a user removes it from `programs.vscode.extensions`.

I think a better solution is to create a `home.file` entry for each extension.

Reply:
How does one uses `home.file` to symlink a directory?

Reply:
Just set the `source` attribute to point to the directory. For example
```
home.file.test.source = ./this/is/a/directory;
```
should work fine.